src/lib/paw-to-openapi-converter/components/parameters-converter.ts

```diff
@@ -1,122 +1,125 @@
+      this.request.variables.forEach((variable) => {
         if (variable) {
-          let newParam: OpenAPI.ParameterObject = {
-            name: variableName,
-            in: "path",
+          const newParam: OpenAPI.ParameterObject = {
+            name: variable.name,
+            in: 'path',
             required: true,
-          }
+          };
 
-          const variableValue = variable.getCurrentValue()
+          const variableValue = variable.getCurrentValue();
 
           if (!variable.required) {
-            newParam.example = { //just to inform Paw while importing back that exported file
+            newParam.example = { // just to inform Paw while importing back that exported file
               summary: NonRequiredLabel,
-              value: true
-            }
+              value: true,
+            };
           }
 
           newParam.schema = {
             type: 'string',
-            default: variableValue ?? ''
-          }
+            default: variableValue ?? '',
+          };
 
-          this.parameters.push(newParam)
+          this.parameters.push(newParam);
         }
-      })
+      });
     }
   }
 
   private parseHeaders(): void {
-    this.request.headersNames.forEach((headerName: string) => {
+    Object.entries(this.request.headers).forEach(([headerName, headerValue]) => {
       if (headerName.toLowerCase() === 'cookie') {
-        this.parseCookies((this.request.headers[headerName] ?? '') as string)
+        this.parseCookies((headerValue ?? '') as string);
       } else {
         const schema: OpenAPI.SchemaObject = {
           type: 'string',
-          default: this.request.headers[headerName] ?? ''
-        }
+          default: headerValue ?? '',
+        };
 
         if (headerName.toLowerCase() === 'content-type' && schema.default !== '') {
           if (schema.default.toLowerCase().indexOf('application/json') >= 0) {
-            this.bodyContentType = 'application/json'
+            this.bodyContentType = 'application/json';
           } else if (schema.default.toLowerCase().indexOf('application/xml') >= 0) {
-            this.bodyContentType = 'application/xml'
+            this.bodyContentType = 'application/xml';
           } else if (schema.default.toLowerCase().indexOf('application/x-www-form-urlencoded') >= 0) {
-            this.bodyContentType = 'application/x-www-form-urlencoded'
-          } else if (schema.default.toLowerCase().indexOf('application/form-data') >= 0) {
-            this.bodyContentType = 'application/form-data'
+            this.bodyContentType = 'application/x-www-form-urlencoded';
+          } else if (schema.default.toLowerCase().indexOf('multipart/form-data') >= 0) {
+            this.bodyContentType = 'multipart/form-data';
           }
         }
 
         this.parameters.push({
           name: headerName,
-          in: "header",
-          schema
-        })
+          in: 'header',
```

Comment:
Let me provide conclusions from my debugging.

_Test case:_
<img width="515" alt="+ POST https example com Path variable" src="https://user-images.githubusercontent.com/69789351/92101912-1c355000-edde-11ea-95e8-cfac4a0d15f9.png">
<img width="518" alt="POST httpsexample comPath variable" src="https://user-images.githubusercontent.com/69789351/92101931-222b3100-edde-11ea-942f-1940b3acfad6.png">
<img width="509" alt="POST httpsexample comPath variable" src="https://user-images.githubusercontent.com/69789351/92101938-248d8b00-edde-11ea-95e5-25a2bbf21239.png">

_Problems:_
I haven’t noticed that headers and query parameters are also stored in „request.variables” as seen on first screenshot. They goes there after adding descriptions which I haven’t done and that is why I missed that issues during development. I thought that „request.variables” only stores path variables and that is why duplicates of headers and query params were added in shown test case.


**How to solve?**


_What data I have during generating output:_
1. request.urlParameters / request.urlParameterNames - which stores values/names of query params
2. request.headers - which stores key<->value map of headers
3. request.variables - which stores information about all - query params, headers and path variables

_What I need to actually output is:_
Put urlParameters, headers and path variables into OpenApi’s operations.parameters with correct „in” param that describes which one is it.

_What problem needs to be solved:_
Path variables do not have separate place in Paw’s request or do not have any „type” or „in” param to be able to distinguish them from other objects (headers, query params) - the request.variables type is defined [here](https://github.com/luckymarmot/Paw-OpenAPI3Generator/blob/74ca00f7904e6631e688bfeaafb2c7265e6181a3/src/types-paw-api/paw.d.ts#L336)

_Propose of solution:_
1. First copy request.variables to local variable (let's name it $requestVariables)
2. Then parse headers and query params  using request.urlParameter(Names) and request.headers
3. While parsing headers and query params try to fetch attached variable to each one For example using [code snippet](https://github.com/luckymarmot/Paw-PostmanCollectionV2Generator/blob/8a153f526ddd37717eaf0d8fe819f014ad866cd0/src/lib/convertHeaders.ts#L11) from Postman Generator (this snippet is for headers).
4. If variable for header or query param will be found, remove it from $requestVariables
5. Parse all $requestVariables that are still left and treat them as path variables.

@mittsh Please take a look at that note and confirm if that way of solving this problem seems ok for You or maybe if You have any additional concerns about that.

Reply:
Ok I got another case that is related to this problem:
![image](https://user-images.githubusercontent.com/69789351/92108979-8226d500-ede8-11ea-998e-21994684224e.png)

_Problems:_
1. The URL in PawRequest has already replaced values, so e.g request.urlBase returns in above example `http://developers.uspto.gov/` instead of `{scheme}://{subdomain}upsto.gov/`. It can make it impossible to distinguish if both of those variables will have same value e.g. 'http'.
2. That variables are also stores in request.variables, but that seems fine since they should be treated same as path variables in example from previous comment.


**How to solve?**


_What data I have during output:_
1. request.urlBase which contain already replaced variables
2. $requestVariables (from previous comment) that should store all path variables left to be processed

_What problems needs to be solved:_
1. While parsing path parameters they needs to be additionally splitted:
a. path params that are in hostname part of URL should be passed to OpenAPI server variables
b. path params that are outside hostname pare of URL should be passed to OpenAPI request variables

_Propose of solution:_
Unfortunatelly for now I have no proposal for that because of that URL already have replaced variables which complicates that situation. For now I'm working on importer [related case](https://github.com/luckymarmot/Paw-OpenAPI3Importer/pull/1#discussion_r478450156) and maybe I'll come up with some solution while working there